pkg/model/initializers.go

```diff
@@ -159,6 +189,21 @@ func (ml *ModelLoader) grpcModel(backend string, o *Options) func(string, string
 			}
 		} else {
 			grpcProcess := backendPath(o.assetDir, backend)
+
+			// for llama-cpp, check CPU capabilities and load the appropriate variant
+			if backend == LLamaCPP {
+				if cpu.X86.HasAVX2 {
```

Comment:
a small nit here: we do have already a wrapper around sys info (used now for debugging) https://github.com/mudler/LocalAI/blob/b4cb22f4449cbc587bc9038a3b6a1f78f84fb8f5/pkg/xsysinfo/cpu.go#L36 , would be nice to be consistent and use that one all around as it has wrappers/logic already to deal with CPU flags